### PR TITLE
New method + overloads Logger.setAsyncContext()

### PR DESCRIPTION
**Pull request by jongpie**
### Core Unlocked Package Changes

- Added new global method (and overloads) `Logger.setAsyncContext()`. There are 4 overloads
   ```apex
    void setAsyncContext(Database.BatchableContext batchableContext) {}
    void setAsyncContext(System.FinalizerContext finalizerContext) {}
    void setAsyncContext(System.QueueableContext queueableContext) {}
    void setAsyncContext(System.SchedulableContext schedulableContext) {}
    ```
- Added new fields on `LogEntryEvent__e` and `Log__c`
  | New Fields | Description |
  | ----------- | ----------- |
  |`LogEntryEvent__e.AsyncContextType__c` and `Log__c.AsyncContextType__c` | The interface of the platform-provided context. Current values are `Database.BatchableContext`, `System.FinalizerContext`, `System.QueueableContext`, and `System.SchedulableContext`. |
  |`LogEntryEvent__e.AsyncContextParentJobId__c` and `Log__c.AsyncContextParentJobId__c` | The ID of the parent job (`Schema.AsyncApexJob`) that initiated the async transaction. Used for `Database.Batchable`, `System.FinalizerContext`, and `System.QueueableContext`. |
  |`LogEntryEvent__e.AsyncContextChildJobId__c` and `Log__c.AsyncContextChildJobId__c` | The ID of the child job (`Schema.AsyncApexJob`) that initiated the async transaction. Used for `Database.Batchable`. |
  |`LogEntryEvent__e.AsyncContextTriggerId__c` and `Log__c.AsyncContextTriggerId__c` | The ID of the cron trigger (`Schema.CronTrigger`) that initiated the async transaction. |
- Updated FlexiPage `LogRecordPage` to conditionally show a new page section, "Async Context Details", which _also_ conditionally shows the 4 new fields

image

- Updated several async classes to leverage the new method `Logger.setAsyncContext()`
- Eliminated the helper method `LoggerMockDataCreator.createBatchableContext()` - the underlying mock batchable context class, as well as some new similar mocks for `Finalizer`, `Queueable, and `Schedulable`, are now public and can be instantiated directly
  - Having the extra static methods was adding a lot of unnecessary complication & redundant code to expose static methods for each constructor

### Pipeline Changes

- Upgraded `sf` cli (again) to try to resolve some packaging issues in the pipeline (again)

### CODE REVIEW

**Comment by jongpie** [comment]
@jamessimone any chance you have some time this week to do a code review on this? I think I've finally fixed all of the issues we chatted about earlier, and I'd love to get your thoughts on this (both in terms of the usefulness of the enhancement, as well as the implementation of the enhancement).

File: nebula-logger/core/main/log-management/flexipages/LogRecordPage.flexipage-meta.xml [hunk 1]
```diff
@@ -525,6 +525,96 @@
...
+            <fieldInstance>
+                <fieldInstanceProperties>
+                    <name>uiBehavior</name>
+                    <value>readonly</value>
+                </fieldInstanceProperties>
+                <fieldItem>Record.AsyncContextTriggerId__c</fieldItem>
+                <identifier>RecordAsyncContextTriggerId_cField</identifier>
+                <visibilityRule>
+                    <criteria>
+                        <leftValue>{!Record.AsyncContextChildJobId__c}</leftValue>
```
Comment (jamessimone) [inline nebula-logger/core/main/log-management/flexipages/LogRecordPage.flexipage-meta.xml:585]: shouldn't this be `{!Record.AsyncContextChildJobId__c}` ?

**Review by jamessimone — APPROVED** [review]
@jongpie I think it looks great!

File: nebula-logger/core/main/log-management/flexipages/LogRecordPage.flexipage-meta.xml (hunk 1 above)
Comment (jongpie) [reply nebula-logger/core/main/log-management/flexipages/LogRecordPage.flexipage-meta.xml:585]: Yeah, great catch! I'll fix this

**Comment by jamessimone** [comment]
🎉🎉🎉

**Pull request merged**